connect/json/src/test/java/org/apache/kafka/connect/json/JsonConverterTest.java

```diff
@@ -252,6 +253,27 @@ public void decimalToConnectOptionalWithDefaultValue() {
         assertEquals(reference, schemaAndValue.value());
     }
 
+    @Test
```

Comment:
IIUC, the only existing test for deserializing a JSON float verified this only for the schema case, and there was no test that verified the behavior for the no schema case.

What do you think about adding more unit tests that check serialization and deserialization for all of the Connect data types?

Reply:
It looks like we have tests for all of the primitive types (e.g. `floatToConnect` for the schema case and `nullSchemaPrimitiveToConnect` for the schemaless test). I hope those would be sufficient to catch any regression there.

To be safe, I've also made sure that only the JsonConverter uses the new deserializer feature to deserialize floats big decimals.